node/hack/bridge-value-tools/main.go

```diff
@@ -0,0 +1,243 @@
+	"io/ioutil"
+	"math"
+	"math/big"
+	"os"
+	"strconv"
+
+	"golang.org/x/text/language"
+	"golang.org/x/text/message"
+)
+
+const ACCOUNTANT_DUMP = "data/accountant.json"
+const TOKEN_ALLOW_LIST = "data/token-allowlist.json"
+const TOKEN_METADATA = "data/token_metadata.json"
+
+type (
+	// Layout of the config data for each token
+	tokenConfigEntry struct {
+		chain       uint16
+		addr        string
+		symbol      string
+		coinGeckoId string
+		decimals    int64
+		price       float64
+	}
+)
+
+func tokenList() []tokenConfigEntry {
+	return append(manualTokenList(), generatedMainnetTokenList()...)
+}
+
+type Account struct {
+	Key     AccountKey `json:"key"`
+	Balance string     `json:"balance"`
+}
+
+type AccountKey struct {
+	ChainId      int    `json:"chain_id"`
+	TokenChain   int    `json:"token_chain"`
+	TokenAddress string `json:"token_address"`
+}
+
+type TokenEntry struct {
+	price          float64
+	decimalDivisor *big.Int
+	symbol         string
+	coinGeckoId    string
+}
+
+// TokenAllowList maps chainId -> (nativeAddress -> coingeckoId)
+type TokenAllowList map[string]map[string]string
+
+type TokenMetadata []TokenMetadataEntry
+type TokenMetadataEntry struct {
+	TokenChain    string `json:"token_chain"`
+	TokenAddress  string `json:"token_address"`
+	NativeAddress string `json:"native_address"`
+}
+
+// NativeAddressLookupTable maps strings "chainId/WhTokenAddress" -> string nativeAddress
+type NativeAddressLookupTable map[string]string
+
+type TokenFilter struct {
+	nativeAddressTbl NativeAddressLookupTable
+	tal              TokenAllowList
+}
+
+func NewTokenFilter() *TokenFilter {
+	// Read Token Allow List
+
+	jsonFileTAL, err := os.Open(TOKEN_ALLOW_LIST)
+	// if we os.Open returns an error then handle it
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	byteValueTAL, _ := ioutil.ReadAll(jsonFileTAL)
+	var tal = TokenAllowList{}
+	json.Unmarshal(byteValueTAL, &tal)
+
+	// Read Token MetaData
+
+	jsonFileTM, err := os.Open(TOKEN_METADATA)
+	// if we os.Open returns an error then handle it
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	byteValueTM, _ := ioutil.ReadAll(jsonFileTM)
+	var tm = TokenMetadata{}
+	var nativeToken = NativeAddressLookupTable{}
+	json.Unmarshal(byteValueTM, &tm)
+
+	for _, tme := range tm {
+		tokenStr := fmt.Sprintf("%s/%s", tme.TokenChain, tme.TokenAddress)
+		//fmt.Printf("%s: %s\n", tokenStr, tme.NativeAddress)
+		nativeToken[tokenStr] = tme.NativeAddress
+	}
+
+	return &TokenFilter{
+		nativeAddressTbl: nativeToken,
+		tal:              tal,
+	}
+}
+
+func (f *TokenFilter) isAllowed(chainId int, whTokenAddress string) bool {
+	// find nativeAddress
+	tokenStr := fmt.Sprintf("%d/%s", chainId, whTokenAddress)
+	nativeAddress, ok := f.nativeAddressTbl[tokenStr]
+	if !ok {
+		return false
+	}
+
+	fmt.Sprintf("Native address for tokenStr: %s\n", nativeAddress)
+
+	ce, ok := f.tal[strconv.Itoa(chainId)]
+	if !ok {
+		return false
+	}
+
+	_, ok = ce[nativeAddress]
+
+	if !ok {
+		return false
+	}
+	return true
```

Comment:
Exactly the same thing:
```suggestion
	return ok
```